js/main.js

```diff
@@ -154,61 +157,78 @@ class App extends React.Component {
     return pathname;
   };
 
-  addPaneBatched = (pane) => {
-    if (!this._timeoutID) {
-      this._timeoutID = setTimeout(this.processBatchedPanes, 100);
+  // ------------------ //
+  // batched processing //
+  // ------------------ //
+
+  // store pane to be processed
+  const addPaneBatched = (pane) => {
+    if (!_timeoutID.current) {
+      _timeoutID.current = setTimeout(processBatchedPanes, 100);
     }
-    this._pendingPanes.push(pane);
+    _pendingPanes.current.push(pane);
   };
 
-  processBatchedPanes = () => {
-    let newPanes = Object.assign({}, this.state.panes);
-    let newLayout = this.state.layout.slice();
+  // run processing on queue
+  const processBatchedPanes = () => {
+    // wait until app is mounted
+    if (!mounted.current) {
+      // console.log('not processing');
+      _timeoutID.current = setTimeout(processBatchedPanes, 100);
+      return;
+    }
+    let newPanes = Object.assign({}, storeData.panes);
+    let newLayout = storeData.layout.slice();
 
-    this._pendingPanes.forEach((pane) => {
-      this.processPane(pane, newPanes, newLayout);
+    _pendingPanes.current.forEach((pane) => {
+      processPane(pane, newPanes, newLayout);
     });
 
-    this._pendingPanes = [];
-    this._timeoutID = null;
+    _pendingPanes.current = [];
+    _timeoutID.current = null;
 
-    this.setState({
+    setStoreData((prev) => ({
+      ...prev,
       panes: newPanes,
       layout: newLayout,
-    });
+    }));
   };
 
-  processPane = (newPane, newPanes, newLayout) => {
+  // process single pane
+  const processPane = (newPane, newPanes, newLayout) => {
     let exists = newPane.id in newPanes;
     newPanes[newPane.id] = newPane;
 
+    consistent_pane_copy.current[newPane.id] = deepcopy(newPane);
```

Comment:
does it change behavior that this deepcopy is made before modifications happening below in the `else`?

Reply:
I moved that line up here, because there has been the a duplicate at the start of the condition (right after)
```
if (!exists)
```
and also in the else branch, however at the end there.

Functionality-wise, there should be no difference in the order, as `newPane` is not changed in this function (`processPane`) at all. Just the layout (`newLayout`) gets an update, but is not affected by `consistent_pane_copy`.